pkg/inject/envoy.go

```diff
@@ -249,6 +259,45 @@ func (m *envoyMutator) getCustomEnv(pod *corev1.Pod) (map[string]string, error)
 	return customEnv, nil
 }
 
+type customEnvJsonType []map[string]string
+
+func (c *customEnvJsonType) UnmarshalJSON(data []byte) error {
```

Comment:
Need we make a valid check for bytes before unmarshal if the input is empty or nil?

Reply:
It will throw malformed error. so, I didn't make a check for that condition.

Reply:
Thanks. Good to know